pkgs/development/haskell-modules/make-package-set.nix

```diff
@@ -161,13 +161,15 @@ in package-set { inherit pkgs stdenv callPackage; } self // {
       filter = path: type:
                  pkgs.lib.hasSuffix "${name}.cabal" path ||
                  baseNameOf path == "package.yaml";
-      expr = self.haskellSrc2nix {
+      expr = self.haskellSrc2nix ({
         inherit name;
         src = if pkgs.lib.canCleanSource src
                 then pkgs.lib.cleanSourceWith { inherit src filter; }
               else src;
-      };
-    in overrideCabal (callPackageKeepDeriver expr args) (orig: {
+      } // haskellSrc2nixArgs);
+      callPackageArgs = filterAttrs (name: v: name != "extraCabal2nixOptions") args;
+      haskellSrc2nixArgs = filterAttrs (name: v: name == "extraCabal2nixOptions") args;
```

Comment:
You could just `inherit` it in the definition of `expr`.

Reply:
Yes, I have thought of that, but we'd have to duplicate the definition of the default argument. I think duplication of definitions is worse than a slightly awkward but clear expression.